tutorials/genomic_alignment/genomic_alignment.py

```diff
@@ -0,0 +1,444 @@
+    read2: FlyteFile | None = None
+
+    def get_read_fnames(self):
+        return (
+            f"{self.sample}_1.fastq.gz",
+            f"{self.sample}_2.fastq.gz",
+        )
+
+    def get_report_fname(self):
+        return f"{self.sample}_fastq-filter-report.json"
+
+    @classmethod
+    def from_remote(cls, urls: List[str]):
+        dl_loc = Path("/tmp/reads")
+        dl_loc.mkdir(exist_ok=True, parents=True)
+
+        samples = {}
+        for fp in [fetch_file(url, dl_loc) for url in urls]:
+            sample = fp.stem.split("_")[0]
+
+            if sample not in samples:
+                samples[sample] = Reads(sample=sample)
+
+            if ".fastq.gz" in fp.name or "fasta" in fp.name:
+                mate = fp.name.strip(".fastq.gz").strip(".filt").split("_")[-1]
+                if "1" in mate:
+                    samples[sample].read1 = FlyteFile(path=str(fp))
+                elif "2" in mate:
+                    samples[sample].read2 = FlyteFile(path=str(fp))
+
+        return list(samples.values())
+
+
+# Finally, we define an `Alignment` data class to represent an alignment file and its associated
+# sample, format, index, and the tool used for the alignment. Alignments are the result of
+# mapping the reads back to the reference genome. This is necessary to perform downstream analyses
+# such as variant calling, gene expression quantification, and more.
+
+
+@dataclass
+class Alignment(DataClassJSONMixin):
+    """
+    Represents a alignment file and its associated sample.
+
+    Attributes:
+        sample (str): The name or identifier of the sample to which the alignment file belongs.
+        aligner (str): The name of the aligner used to generate the alignment file.
+        format (str): The format of the alignment file (e.g., SAM, BAM).
+        alignment (FlyteFile): A FlyteFile object representing the path to the alignment file.
+        alignment_idx (FlyteFile): A FlyteFile object representing an alignment index file.
+    """
+
+    sample: str
+    aligner: str
+    format: str | None = None
+    alignment: FlyteFile | None = None
+    alignment_idx: FlyteFile | None = None
+
+    def get_alignment_fname(self):
+        return f"{self.sample}_{self.aligner}_aligned.{self.format}"
+
+
+# ## Tasks
+#
+# We define a series of tasks to perform the following operations:
+# 1. Fetch assets from remote URLs
+# 2. Perform quality filtering and preprocessing using FastP
+# 3. Generate Bowtie2 index files from a reference genome
+# 4. Perform alignment using Bowtie2 on a filtered sample
+#
+# The first task is quite simple, it simply calls the `from_remote`
+# methods on both the `Reference` and `Reads` classes. It will also
+# cache these assets so they won't need to be re-downloaded. This isn't
+# as important with the small files we're working with here, but can be
+# crucial when working with large reference genomes and sequencing data.
+
+
+@task(container_image=main_img, cache=True, cache_version="4")
```

Comment:
I feel like using `cache_version="4"` shows that four attempts were made when writing the tutorial. Not that this is bad at all, but we could use `cache_version="1.0"` similar to the docs [here](https://docs.union.ai/byoc/core-concepts/caching#example).
```suggestion
@task(container_image=main_img, cache=True, cache_version="1.0")
```

Reply:
Yeah wasn't sure how to handle this, but that works!